File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/variables/types/NumberType.java
package ch.rmy.android.http_shortcuts.variables.types;

import android.text.InputType;

import com.afollestad.materialdialogs.MaterialDialog;

import org.jdeferred.Deferred;

import ch.rmy.android.http_shortcuts.realm.Controller;
import ch.rmy.android.http_shortcuts.realm.models.Variable;

public class NumberType extends TextType {

    @Override
    public void setupDialog(Controller controller, Variable variable, MaterialDialog.Builder builder, Deferred<String, Void, Void> deferredValue) {
        super.setupDialog(controller, variable, builder, deferredValue);
        builder.inputType(InputType.TYPE_CLASS_NUMBER);
    }

}

File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/variables/types/NumberType.kt
package ch.rmy.android.http_shortcuts.variables.types

import android.content.Context
import android.text.InputType
import ch.rmy.android.http_shortcuts.realm.Controller
import ch.rmy.android.http_shortcuts.realm.models.Variable
import ch.rmy.android.http_shortcuts.utils.showIfPossible
import org.jdeferred2.Deferred

internal class NumberType : TextType() {

    override fun createDialog(context: Context, controller: Controller, variable: Variable, deferredValue: Deferred<String, Unit, Unit>): () -> Unit {
        val builder = createDialogBuilder(context, variable, deferredValue)
                .toDialogBuilder()
                .input(null, if (variable.rememberValue) variable.value else "") { _, input ->
                    deferredValue.resolve(input.toString())
                    controller.setVariableValue(variable.id, input.toString()).subscribe()
                }
                .inputType(InputType.TYPE_CLASS_NUMBER or InputType.TYPE_NUMBER_FLAG_DECIMAL or InputType.TYPE_NUMBER_FLAG_SIGNED)
        return {
            builder.showIfPossible()
        }
    }

}
